Source/CombatExtended/CombatExtended/Jobs/JobDriver_Hunt.cs

```diff
@@ -196,10 +195,57 @@ Toil GotoCastPosition(TargetIndex targetInd, bool closeIfDowned = false)
             return toil;
         }
 
-        //Fit for an attack range per body size curve.
-        public static float HuntRangePerBodysize(float x, float executionRange, float gunRange)
+        public static float GetOptimalHuntRange(Pawn hunter, Pawn victim)
         {
-            return Mathf.Min(Mathf.Clamp(1 + 20 * (1 - Mathf.Exp(-0.65f * x)), executionRange, 20), gunRange);
+            var curJob = hunter.CurJob;
+            var victimProps = victim.RaceProps;
+            
+            if (victim.Downed)
+            {
+                // TODO: How to detect boombats?
+                return Mathf.Min(curJob.verbToUse.verbProps.range, victimProps.executionRange);
+            }
+            
+            var normalRange = HuntRangePerBodysize(victimProps.baseBodySize, victimProps.executionRange, curJob.verbToUse.verbProps.range);
+
+            if (victimProps.manhunterOnDamageChance > 0)
+            {
+                // NOTE: I assumed 2 is a good number to consider for aim time of hunting weapons (since hunting weapons are likely to be sniper rifles)
+                // We can also get aim time from the actual weapon that the hunter is using
+                // but I didn't see the need.
+                const int aimTime = 2;
```

Comment:
This is fine, since it's the longest anything is likely to take, so over-estimating serves to make the hunter get closer. If they aren't using a sniper weapon, they likely have more firepower available, so being closer is fine.